Frameworks/Foundation/NSMutableArray.mm

```diff
@@ -557,11 +557,23 @@ - (void)replaceObjectsInRange:(NSRange)aRange withObjectsFromArray:(NSArray*)oth
 }
 
 /**
- @Status Caveat
- @Notes Options are currently ignored
+ @Status Interoperable
+ @Notes NSSortConcurrent is ignored, but this is compatible with reference documentation
 */
 - (void)sortWithOptions:(NSSortOptions)opts usingComparator:(NSComparator)cmptr {
-    [self sortUsingFunction:CFNSBlockCompare context:cmptr];
+    if (NSSortStable) {
+        __block NSArray* original = [[self mutableCopy] autorelease];
```

Comment:
> __block NSArray* original = [[self mutableCopy] autorelease]; [](start = 7, length = 62)

use a dictionary instead of an array to make this faster.